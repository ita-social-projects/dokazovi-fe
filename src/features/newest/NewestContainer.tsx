import React, { useEffect, useState } from 'react';
import { useStyles } from './NewestContainer.style';
import { LoadingStatusEnum, LoadingStatusType } from '../../old/lib/types';
import { NewestPostsList } from './newestPostsList/NewestPostsList';
import {
  NewestPostResponseType,
  NewestTypeEnum,
} from '../../old/lib/utilities/API/types';
import { getNewestPosts } from '../../old/lib/utilities/API/api';

export const NewestContainer: React.FC = () => {
  const classes = useStyles();
  const [postsSets, setPostsSets] = useState<NewestPostResponseType[]>([]);
  const [loadingStatus, setLoadingStatus] = useState<LoadingStatusType>(
    LoadingStatusEnum.pending,
  );

  useEffect(() => {
    getNewestPosts()
      .then((res): void => {
        setPostsSets(res.data.content);
        setLoadingStatus(LoadingStatusEnum.succeeded);
      })
      .catch((): void => {
        setLoadingStatus(LoadingStatusEnum.failed);
      });
  }, []);

  return (
    <>
      <div className={classes.container}>
        <NewestPostsList
          loadingStatus={loadingStatus}
          postsListTitle="Думки експертів"
          postsListPath="materials?origins=1"
          postsList={
            postsSets[NewestTypeEnum.EXPERT_OPINION] &&
            postsSets[NewestTypeEnum.EXPERT_OPINION].postDTOS
          }
        />
        <NewestPostsList
          loadingStatus={loadingStatus}
          postsListTitle="Медитека"
          postsListPath="materials?origins=2"
          postsList={
            postsSets[NewestTypeEnum.MEDIA] &&
            postsSets[NewestTypeEnum.MEDIA].postDTOS
          }
        />
        <NewestPostsList
          loadingStatus={loadingStatus}
          postsListTitle="Переклади"
          postsListPath="materials?origins=3"
          postsList={
            postsSets[NewestTypeEnum.TRANSLATION] &&
            postsSets[NewestTypeEnum.TRANSLATION].postDTOS
          }
        />
        <NewestPostsList
          loadingStatus={loadingStatus}
          postsListTitle="Відео"
          postsListPath="materials?types=2"
          postsList={
            postsSets[NewestTypeEnum.VIDEO] &&
            postsSets[NewestTypeEnum.VIDEO].postDTOS
          }
        />
      </div>
    </>
  );
};
