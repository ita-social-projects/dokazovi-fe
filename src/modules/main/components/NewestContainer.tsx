import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Container, Grid, Button, Typography } from '@material-ui/core';

import { IMainState } from '../store/mainReducer';
import { IRootState } from '../../../store/rootReducer';
import { loadNewestThunk } from '../store/actions';
import PostPreviewCard from '../../../lib/components/PostPreview/PostPreviewCard';
import { useStyles } from './styles/NewestContainer.style';
import BorderBottom from '../../../lib/components/Border';

const NewestContainer: React.FC = () => {
  const classes = useStyles();

  const dispatch = useDispatch();
  const setNewest = () => dispatch(loadNewestThunk());

  const { newestPosts, meta } = useSelector<IRootState, IMainState['newest']>(
    (state) => {
      return state.main.newest;
    },
  );

  useEffect(() => {
    setNewest();
  }, []);

  return (
    <>
      <Container>
        <Typography variant="h4">Найновіше</Typography>
        <Grid container spacing={2} direction="row" alignItems="center">
          {newestPosts.map((post) => (
            <Grid item xs={12} lg={4} md={6} key={post.author?.phone}>
              <PostPreviewCard data={post} />
            </Grid>
          ))}
        </Grid>

        <Grid
          container
          direction="column"
          alignItems="center"
          className={classes.showMore}
        >
          {meta.showMore ? (
            <Button variant="contained" onClick={setNewest}>
              Більше матеріалів
            </Button>
          ) : (
            <span>Більше нових матеріалів немає</span>
          )}
        </Grid>
        <BorderBottom />
      </Container>
    </>
  );
};

export default NewestContainer;
