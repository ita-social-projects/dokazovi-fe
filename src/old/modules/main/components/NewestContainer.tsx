import React, { useEffect } from 'react';
import { useSelector } from 'react-redux';
import { useActions } from '../../../lib/hooks/useActions';
import { RootStateType } from '../../../store/rootReducer';
import { fetchNewestPosts } from '../store/mainSlice';
import { useStyles } from '../styles/NewestContainer.style';
import PostsList from '../../../lib/components/Posts/PostsList';
import { LoadingStatusEnum } from '../../../lib/types';
import { selectPostsByIds } from '../../../store/selectors';
import LoadingContainer from '../../../lib/components/Loading/LoadingContainer';

const NewestContainer: React.FC = () => {
  const classes = useStyles();

  const {
    newestPostIds,
    meta: { loading },
  } = useSelector((state: RootStateType) => state.main.newest);
  const newestPosts = selectPostsByIds(newestPostIds);

  const actionFetchNewestPosts = useActions(fetchNewestPosts);

  useEffect(() => {
    actionFetchNewestPosts();
  }, []);

  return (
    <div className={classes.container}>
      {loading === LoadingStatusEnum.pending ? (
        <LoadingContainer loading={loading} expand />
      ) : (
        <>
          <PostsList postsList={newestPosts} />
          <LoadingContainer loading={loading} />
        </>
      )}
    </div>
  );
};

export default NewestContainer;
