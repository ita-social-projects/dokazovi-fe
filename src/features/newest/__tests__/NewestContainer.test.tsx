import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { NewestContainer } from '../NewestContainer';
import { mockNewestContent } from '../mockNewestContent';

jest.mock('../../../old/lib/utilities/API/api', () => ({
  getNewestPosts: () => Promise.resolve(mockNewestContent),
}));

describe('NewestContainer component', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <NewestContainer />
      </MemoryRouter>,
    );

    expect(container).toMatchSnapshot();
  });
});
