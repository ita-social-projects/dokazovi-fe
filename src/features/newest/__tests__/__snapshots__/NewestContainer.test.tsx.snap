// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewestContainer component should match snapshot 1`] = `
<div>
  <div
    class="NewestContainer-container-1"
  >
    <div
      class="NewestPostsList-container-2"
    >
      <a
        href="/materials?origins=1"
      >
        <div
          class="NewestPostsList-header-3"
        >
          <span
            class="MuiTypography-root NewestPostsList-postsType-4 MuiTypography-h5"
          >
            Думки експертів
          </span>
          <div
            class="NewestPostsList-line-5"
          />
          <div
            class="NewestPostsList-arrow-6"
          >
            <span />
            <span />
            <span />
            <span />
          </div>
        </div>
      </a>
      <div
        class="MuiGrid-root NewestPostsList-grid-7 MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-item"
      >
        <div>
          <div
            class="MuiPaper-root MuiCard-root MediaPostPreviewCard-root-9 MuiPaper-elevation1 MuiPaper-rounded"
            data-testid="skeleton"
          >
            <div
              class="MuiBox-root MuiBox-root-16 MediaPostPreviewCard-header-10 MediaPostPreviewCard-header-15"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-postType-14 MuiSkeleton-pulse"
              />
            </div>
            <div
              class="MuiBox-root MuiBox-root-17 MediaPostPreviewCard-body-11"
            >
              <h3
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textHeader-12 MuiSkeleton-pulse"
              />
              <p
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textBody-13 MuiSkeleton-pulse"
                color="textPrimary"
              />
              <div
                class="MuiBox-root MuiBox-root-18"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  color="textSecondary"
                  style="width: 60%;"
                />
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="MuiPaper-root MuiCard-root MediaPostPreviewCard-root-9 MuiPaper-elevation1 MuiPaper-rounded"
            data-testid="skeleton"
          >
            <div
              class="MuiBox-root MuiBox-root-20 MediaPostPreviewCard-header-10 MediaPostPreviewCard-header-19"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-postType-14 MuiSkeleton-pulse"
              />
            </div>
            <div
              class="MuiBox-root MuiBox-root-21 MediaPostPreviewCard-body-11"
            >
              <h3
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textHeader-12 MuiSkeleton-pulse"
              />
              <p
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textBody-13 MuiSkeleton-pulse"
                color="textPrimary"
              />
              <div
                class="MuiBox-root MuiBox-root-22"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  color="textSecondary"
                  style="width: 60%;"
                />
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="MuiPaper-root MuiCard-root MediaPostPreviewCard-root-9 MuiPaper-elevation1 MuiPaper-rounded"
            data-testid="skeleton"
          >
            <div
              class="MuiBox-root MuiBox-root-24 MediaPostPreviewCard-header-10 MediaPostPreviewCard-header-23"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-postType-14 MuiSkeleton-pulse"
              />
            </div>
            <div
              class="MuiBox-root MuiBox-root-25 MediaPostPreviewCard-body-11"
            >
              <h3
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textHeader-12 MuiSkeleton-pulse"
              />
              <p
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textBody-13 MuiSkeleton-pulse"
                color="textPrimary"
              />
              <div
                class="MuiBox-root MuiBox-root-26"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  color="textSecondary"
                  style="width: 60%;"
                />
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="MuiPaper-root MuiCard-root MediaPostPreviewCard-root-9 MuiPaper-elevation1 MuiPaper-rounded"
            data-testid="skeleton"
          >
            <div
              class="MuiBox-root MuiBox-root-28 MediaPostPreviewCard-header-10 MediaPostPreviewCard-header-27"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-postType-14 MuiSkeleton-pulse"
              />
            </div>
            <div
              class="MuiBox-root MuiBox-root-29 MediaPostPreviewCard-body-11"
            >
              <h3
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textHeader-12 MuiSkeleton-pulse"
              />
              <p
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textBody-13 MuiSkeleton-pulse"
                color="textPrimary"
              />
              <div
                class="MuiBox-root MuiBox-root-30"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  color="textSecondary"
                  style="width: 60%;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="NewestPostsList-container-2"
    >
      <a
        href="/materials?origins=2"
      >
        <div
          class="NewestPostsList-header-3"
        >
          <span
            class="MuiTypography-root NewestPostsList-postsType-4 MuiTypography-h5"
          >
            Медитека
          </span>
          <div
            class="NewestPostsList-line-5"
          />
          <div
            class="NewestPostsList-arrow-6"
          >
            <span />
            <span />
            <span />
            <span />
          </div>
        </div>
      </a>
      <div
        class="MuiGrid-root NewestPostsList-grid-7 MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-item"
      >
        <div>
          <div
            class="MuiPaper-root MuiCard-root MediaPostPreviewCard-root-9 MuiPaper-elevation1 MuiPaper-rounded"
            data-testid="skeleton"
          >
            <div
              class="MuiBox-root MuiBox-root-32 MediaPostPreviewCard-header-10 MediaPostPreviewCard-header-31"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-postType-14 MuiSkeleton-pulse"
              />
            </div>
            <div
              class="MuiBox-root MuiBox-root-33 MediaPostPreviewCard-body-11"
            >
              <h3
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textHeader-12 MuiSkeleton-pulse"
              />
              <p
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textBody-13 MuiSkeleton-pulse"
                color="textPrimary"
              />
              <div
                class="MuiBox-root MuiBox-root-34"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  color="textSecondary"
                  style="width: 60%;"
                />
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="MuiPaper-root MuiCard-root MediaPostPreviewCard-root-9 MuiPaper-elevation1 MuiPaper-rounded"
            data-testid="skeleton"
          >
            <div
              class="MuiBox-root MuiBox-root-36 MediaPostPreviewCard-header-10 MediaPostPreviewCard-header-35"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-postType-14 MuiSkeleton-pulse"
              />
            </div>
            <div
              class="MuiBox-root MuiBox-root-37 MediaPostPreviewCard-body-11"
            >
              <h3
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textHeader-12 MuiSkeleton-pulse"
              />
              <p
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textBody-13 MuiSkeleton-pulse"
                color="textPrimary"
              />
              <div
                class="MuiBox-root MuiBox-root-38"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  color="textSecondary"
                  style="width: 60%;"
                />
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="MuiPaper-root MuiCard-root MediaPostPreviewCard-root-9 MuiPaper-elevation1 MuiPaper-rounded"
            data-testid="skeleton"
          >
            <div
              class="MuiBox-root MuiBox-root-40 MediaPostPreviewCard-header-10 MediaPostPreviewCard-header-39"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-postType-14 MuiSkeleton-pulse"
              />
            </div>
            <div
              class="MuiBox-root MuiBox-root-41 MediaPostPreviewCard-body-11"
            >
              <h3
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textHeader-12 MuiSkeleton-pulse"
              />
              <p
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textBody-13 MuiSkeleton-pulse"
                color="textPrimary"
              />
              <div
                class="MuiBox-root MuiBox-root-42"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  color="textSecondary"
                  style="width: 60%;"
                />
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="MuiPaper-root MuiCard-root MediaPostPreviewCard-root-9 MuiPaper-elevation1 MuiPaper-rounded"
            data-testid="skeleton"
          >
            <div
              class="MuiBox-root MuiBox-root-44 MediaPostPreviewCard-header-10 MediaPostPreviewCard-header-43"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-postType-14 MuiSkeleton-pulse"
              />
            </div>
            <div
              class="MuiBox-root MuiBox-root-45 MediaPostPreviewCard-body-11"
            >
              <h3
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textHeader-12 MuiSkeleton-pulse"
              />
              <p
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textBody-13 MuiSkeleton-pulse"
                color="textPrimary"
              />
              <div
                class="MuiBox-root MuiBox-root-46"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  color="textSecondary"
                  style="width: 60%;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="NewestPostsList-container-2"
    >
      <a
        href="/materials?origins=3"
      >
        <div
          class="NewestPostsList-header-3"
        >
          <span
            class="MuiTypography-root NewestPostsList-postsType-4 MuiTypography-h5"
          >
            Переклади
          </span>
          <div
            class="NewestPostsList-line-5"
          />
          <div
            class="NewestPostsList-arrow-6"
          >
            <span />
            <span />
            <span />
            <span />
          </div>
        </div>
      </a>
      <div
        class="MuiGrid-root NewestPostsList-grid-7 MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-item"
      >
        <div>
          <div
            class="MuiPaper-root MuiCard-root MediaPostPreviewCard-root-9 MuiPaper-elevation1 MuiPaper-rounded"
            data-testid="skeleton"
          >
            <div
              class="MuiBox-root MuiBox-root-48 MediaPostPreviewCard-header-10 MediaPostPreviewCard-header-47"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-postType-14 MuiSkeleton-pulse"
              />
            </div>
            <div
              class="MuiBox-root MuiBox-root-49 MediaPostPreviewCard-body-11"
            >
              <h3
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textHeader-12 MuiSkeleton-pulse"
              />
              <p
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textBody-13 MuiSkeleton-pulse"
                color="textPrimary"
              />
              <div
                class="MuiBox-root MuiBox-root-50"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  color="textSecondary"
                  style="width: 60%;"
                />
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="MuiPaper-root MuiCard-root MediaPostPreviewCard-root-9 MuiPaper-elevation1 MuiPaper-rounded"
            data-testid="skeleton"
          >
            <div
              class="MuiBox-root MuiBox-root-52 MediaPostPreviewCard-header-10 MediaPostPreviewCard-header-51"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-postType-14 MuiSkeleton-pulse"
              />
            </div>
            <div
              class="MuiBox-root MuiBox-root-53 MediaPostPreviewCard-body-11"
            >
              <h3
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textHeader-12 MuiSkeleton-pulse"
              />
              <p
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textBody-13 MuiSkeleton-pulse"
                color="textPrimary"
              />
              <div
                class="MuiBox-root MuiBox-root-54"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  color="textSecondary"
                  style="width: 60%;"
                />
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="MuiPaper-root MuiCard-root MediaPostPreviewCard-root-9 MuiPaper-elevation1 MuiPaper-rounded"
            data-testid="skeleton"
          >
            <div
              class="MuiBox-root MuiBox-root-56 MediaPostPreviewCard-header-10 MediaPostPreviewCard-header-55"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-postType-14 MuiSkeleton-pulse"
              />
            </div>
            <div
              class="MuiBox-root MuiBox-root-57 MediaPostPreviewCard-body-11"
            >
              <h3
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textHeader-12 MuiSkeleton-pulse"
              />
              <p
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textBody-13 MuiSkeleton-pulse"
                color="textPrimary"
              />
              <div
                class="MuiBox-root MuiBox-root-58"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  color="textSecondary"
                  style="width: 60%;"
                />
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="MuiPaper-root MuiCard-root MediaPostPreviewCard-root-9 MuiPaper-elevation1 MuiPaper-rounded"
            data-testid="skeleton"
          >
            <div
              class="MuiBox-root MuiBox-root-60 MediaPostPreviewCard-header-10 MediaPostPreviewCard-header-59"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-postType-14 MuiSkeleton-pulse"
              />
            </div>
            <div
              class="MuiBox-root MuiBox-root-61 MediaPostPreviewCard-body-11"
            >
              <h3
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textHeader-12 MuiSkeleton-pulse"
              />
              <p
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textBody-13 MuiSkeleton-pulse"
                color="textPrimary"
              />
              <div
                class="MuiBox-root MuiBox-root-62"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  color="textSecondary"
                  style="width: 60%;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="NewestPostsList-container-2"
    >
      <a
        href="/materials?types=2"
      >
        <div
          class="NewestPostsList-header-3"
        >
          <span
            class="MuiTypography-root NewestPostsList-postsType-4 MuiTypography-h5"
          >
            Відео
          </span>
          <div
            class="NewestPostsList-line-5"
          />
          <div
            class="NewestPostsList-arrow-6"
          >
            <span />
            <span />
            <span />
            <span />
          </div>
        </div>
      </a>
      <div
        class="MuiGrid-root NewestPostsList-grid-7 MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-item"
      >
        <div>
          <div
            class="MuiPaper-root MuiCard-root MediaPostPreviewCard-root-9 MuiPaper-elevation1 MuiPaper-rounded"
            data-testid="skeleton"
          >
            <div
              class="MuiBox-root MuiBox-root-64 MediaPostPreviewCard-header-10 MediaPostPreviewCard-header-63"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-postType-14 MuiSkeleton-pulse"
              />
            </div>
            <div
              class="MuiBox-root MuiBox-root-65 MediaPostPreviewCard-body-11"
            >
              <h3
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textHeader-12 MuiSkeleton-pulse"
              />
              <p
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textBody-13 MuiSkeleton-pulse"
                color="textPrimary"
              />
              <div
                class="MuiBox-root MuiBox-root-66"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  color="textSecondary"
                  style="width: 60%;"
                />
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="MuiPaper-root MuiCard-root MediaPostPreviewCard-root-9 MuiPaper-elevation1 MuiPaper-rounded"
            data-testid="skeleton"
          >
            <div
              class="MuiBox-root MuiBox-root-68 MediaPostPreviewCard-header-10 MediaPostPreviewCard-header-67"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-postType-14 MuiSkeleton-pulse"
              />
            </div>
            <div
              class="MuiBox-root MuiBox-root-69 MediaPostPreviewCard-body-11"
            >
              <h3
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textHeader-12 MuiSkeleton-pulse"
              />
              <p
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textBody-13 MuiSkeleton-pulse"
                color="textPrimary"
              />
              <div
                class="MuiBox-root MuiBox-root-70"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  color="textSecondary"
                  style="width: 60%;"
                />
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="MuiPaper-root MuiCard-root MediaPostPreviewCard-root-9 MuiPaper-elevation1 MuiPaper-rounded"
            data-testid="skeleton"
          >
            <div
              class="MuiBox-root MuiBox-root-72 MediaPostPreviewCard-header-10 MediaPostPreviewCard-header-71"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-postType-14 MuiSkeleton-pulse"
              />
            </div>
            <div
              class="MuiBox-root MuiBox-root-73 MediaPostPreviewCard-body-11"
            >
              <h3
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textHeader-12 MuiSkeleton-pulse"
              />
              <p
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textBody-13 MuiSkeleton-pulse"
                color="textPrimary"
              />
              <div
                class="MuiBox-root MuiBox-root-74"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  color="textSecondary"
                  style="width: 60%;"
                />
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="MuiPaper-root MuiCard-root MediaPostPreviewCard-root-9 MuiPaper-elevation1 MuiPaper-rounded"
            data-testid="skeleton"
          >
            <div
              class="MuiBox-root MuiBox-root-76 MediaPostPreviewCard-header-10 MediaPostPreviewCard-header-75"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-postType-14 MuiSkeleton-pulse"
              />
            </div>
            <div
              class="MuiBox-root MuiBox-root-77 MediaPostPreviewCard-body-11"
            >
              <h3
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textHeader-12 MuiSkeleton-pulse"
              />
              <p
                class="MuiSkeleton-root MuiSkeleton-text MediaPostPreviewCard-textBody-13 MuiSkeleton-pulse"
                color="textPrimary"
              />
              <div
                class="MuiBox-root MuiBox-root-78"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  color="textSecondary"
                  style="width: 60%;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
